tests/test_svg.py

```diff
@@ -141,53 +141,6 @@ def test_diagram_decorations(
         diag.render("svg")
 
 
-class TestDecoFactory:
-    @pytest.mark.parametrize(
-        "class_",
-        ["LogicalComponentSymbol", "LogicalHumanActorSymbol", "EntitySymbol"],
-    )
-    def test_deco_factory_contains_styling_for_given_styleclass(
-        self, class_: str
-    ):
-        assert class_ in decorations.deco_factories
-
-    def test_deco_factory_returns_symbol_factory_for_given_styleclass(self):
-        deco_factory = decorations.DecoFactory(symbols.port_symbol, ())
-
-        assert decorations.deco_factories["PortSymbol"] == deco_factory
-
-    @pytest.mark.parametrize(
-        "class_", ["ImaginaryClassSymbol", "NothingSymbol"]
-    )
-    def test_deco_factory_logs_error_when_not_containing_given_styleclass(
-        self, caplog, class_: str
-    ):
-        assert class_ not in decorations.deco_factories
-        assert (
-            decorations.deco_factories[class_]
-            is decorations.deco_factories["ErrorSymbol"]
-        )
-        with caplog.at_level(0, logger="decorations"):
-            assert (
-                caplog.messages[-1] == f"{class_} wasn't found in factories."
-            )
-
-    @pytest.mark.parametrize("attr", ["start", "translate", "offsets"])
-    def test_making_linear_gradient_faulty_cases_raise_ValueError(
-        self, attr: str
-    ):
-        params = {
-            "id_": "test",
-            attr: (0, 0, 0),
-        }
-        with pytest.raises(ValueError):
-            symbols._make_lgradient(**params)
-
-    def test_making_linear_gradient_with_translate(self):
-        gradient = symbols._make_lgradient("test", translate=(0, 0))
-        assert gradient.attribs.get("gradientTransform") == "translate(0 0)"
-
-
```

Comment:
I don't like that we remove testing of the icon factory and don't add it again. The Factory moved now into the diagram._icons module. This isn't exposed but still we should test it?!

Reply:
I removed these tests because they weren't really doing anything useful:

- 1 and 2 spot-check the factory registration. Basically useless, considering the number of keys that also should exist but aren't tested. (They really should have tested the decorator behind the factory registration, but that is almost trivial and IMO not worth testing either - you can't even say the regex there is complicated.)
- 3 checks that specific things get logged. Why?
- 4 and 5 were really testing svgwrite and not our code
 
If you have some ideas for tests that make sense in this context, or you disagree on what is worth testing here, let me know tho :)